jerry-core/ecma/base/ecma-helpers-values-collection.cpp

```diff
@@ -38,7 +38,7 @@ ecma_new_values_collection (const ecma_value_t values_buffer[], /**< ecma-values
 {
   JERRY_ASSERT (values_buffer != NULL || values_number == 0);
 
-  const size_t values_in_chunk = sizeof (ecma_collection_chunk_t::data) / sizeof (ecma_value_t);
+  const size_t values_in_chunk = sizeof (((ecma_collection_chunk_t *)0)->data) / sizeof (ecma_value_t);
```

Comment:
@jiangzidong It's not that nice to repeat the sizeof-null-deref logic all over the code. It'd be better to have it in a macro, as suggested in the linked stackoverflow post, e.g.,
```c
#define member_size(type, member) sizeof(((type *)0)->member)
```
(If we choose to use this approach. There are some comments that it is either an undefined behaviour in old C, or needs C99 compiler at least.)